src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts

```diff
@@ -0,0 +1,317 @@
+        autoReplyData: [{ question, answerTemplate: answer.split('\n') }],
+        formData: [
+          {
+            question,
+            answer,
+            answerTemplate: answer.split('\n'),
+            fieldType: 'textfield',
+          },
+        ],
+      })
+    })
+
+    it('should split table answers by newline', () => {
+      const answerArray = [['firstLine\nsecondLine', 'thirdLine\nfourthLine']]
+      const response = generateNewTableResponse({ answerArray })
+
+      const emailData = EmailSubmissionService.createEmailData(
+        [response],
+        new Set(),
+      )
+
+      const question = response.question
+      const answer = answerArray[0].join(',')
+
+      expect(emailData).toEqual({
+        jsonData: [{ question: `[table] ${question}`, answer }],
+        autoReplyData: [{ question, answerTemplate: answer.split('\n') }],
+        formData: [
+          {
+            question: `[table] ${question}`,
+            answer,
+            answerTemplate: answer.split('\n'),
+            fieldType: 'table',
+          },
+        ],
+      })
+    })
+
+    it('should split checkbox answers by newline', () => {
+      const answerArray = ['firstLine\nsecondLine', 'thirdLine\nfourtLine']
+      const response = generateNewCheckboxResponse({ answerArray })
+
+      const emailData = EmailSubmissionService.createEmailData(
+        [response],
+        new Set(),
+      )
+
+      const question = response.question
+      const answer = answerArray.join(', ')
+
+      expect(emailData).toEqual({
+        jsonData: [{ question, answer }],
+        autoReplyData: [{ question, answerTemplate: answer.split('\n') }],
+        formData: [
+          {
+            question,
+            answer,
+            answerTemplate: answer.split('\n'),
+            fieldType: 'checkbox',
+          },
+        ],
+      })
+    })
+
+    it('should prefix verified fields with [verified] only in form data', () => {
+      const response = generateNewSingleAnswerResponse(BasicField.Email, {
+        isUserVerified: true,
+      })
+
+      const emailData = EmailSubmissionService.createEmailData(
+        [response],
+        new Set(),
+      )
+
+      const question = response.question
+      const answer = response.answer
+
+      expect(emailData).toEqual({
+        jsonData: [{ question, answer }],
+        autoReplyData: [{ question, answerTemplate: [answer] }],
+        formData: [
+          {
+            question: `[verified] ${question}`,
+            answer,
+            answerTemplate: [answer],
+            fieldType: 'email',
+          },
+        ],
+      })
+    })
+
+    it('should prefix MyInfo-verified fields with [MyInfo] only in form data', () => {
+      const response = generateNewSingleAnswerResponse(BasicField.ShortText, {
+        myInfo: { attr: MyInfoAttribute.Name },
+      })
+
+      const emailData = EmailSubmissionService.createEmailData(
+        [response],
+        new Set([response._id]),
```

Comment:
can we improve this test case to test for multiple responses? or even add another test case for duplicate myinfo field types to verify the recent bug does not occur